server/runner/taskrun/scheduler.go

```diff
@@ -1023,14 +1028,16 @@ func (s *Scheduler) PatchTaskStatus(ctx context.Context, task *api.Task, taskSta
 		}
 	}
 
-	// Every task in the stage completed means we are moving into a new stage. We need to
-	// 1. cancel external approval.
+	// Every task in the stage completed means we are moving into a new stage.
```

Comment:
Can we refactor this and move it to a separate function?

OnStageDoneHook() something like this?